### ISSUE TITLE
double quotes suddenly escaped.

### ISSUE THREAD

**Issue by davewood** [garu/Data-Printer#165]
after updating Data::Printer from 0.40 to 1.000004 the testsuite of my app fails.

strings that used to look like `'["a","b","c"],["d","e","f"]'`

now look like this `'[\"a\",\"b\",\"c\"],[\"d\",\"e\",\"f\"]'`

data gets passed around extensively and I could't tfind the exact spot where the problem occurs. All I know so far is that updating Data::Printer breaks the tests.

**Comment by davewood** [comment]
```
use Data::Printer {
    return_value   => 'dump',
    indent         => 2,
    hash_separator => ' => ',
    index          => 0,
    colored        => 0,
};

my $pl = { 'default' => { 'foo' => '["a"]' } };
my $str = p($pl);
die $str;
```

0.40
```
\ {
  default => {
    foo => "["a"]"
  }
} at foo.pl line 19.
```

1.000004
```
{
  default => {
    foo => "[\"a\"]"
  }
} at foo.pl line 19.
```

**Comment by davewood** [comment]
https://metacpan.org/dist/Data-Printer/changes#L69

i guess this is the culprit?

However, I need the old behaviour. What can I do?

**Comment by garu** [comment]
Hi, @davewood! First of all, thank you very much for using Data::Printer and for taking the time to submit and review this issue - I really appreciate it.

The rationale behind the change was that the surrounding quotes are not part of the string and, as such, should be differentiated somehow. The most obvious way is, of course, to escape it when the content has the same character.

However, upon reflexion, I think this goes against Data::Printer's mission of making it easier for humans, not machines, to read  data dumps. So I'm going to release a fix on 1.1.0 that will let you restore the old behavior.

Until then, sadly, I believe you only have three options:

* Downgrade to 0.40 until 1.1.0 is released; or

* Manually change your local copy of Data::Printer [here](https://github.com/garu/Data-Printer/blob/main/lib/Data/Printer/Filter/SCALAR.pm#L78) to say `return $text;` instead of a blank line; or

* Update your tests.

Note that if you set the ```scalar_quotes``` option to a single quote, it will not escape the double-quotes:

```perl
    my $x = '["a"]';
    p $x;  # "[\"a\"]"
    p $x, scalar_quotes => q(');  # '["a"]'
```

But you would probably still have to update your test cases, so it's not really a fix.

I'll do my best to release a new version as soon as possible.

Thank you again for bringing this to my attention.

Cheers!

**Comment by davewood** [comment]
after running the production code with the newest version of Data::Printer 1.000004 and everyhting behaving as expected I came to the conclusion that my testsuite wasnt written properly. I just tested what Data::Printer returned and now that the return value is different i might aswell test the new format.

as far as im concerned im happy with the new behaviour

cheers
